crates/scroll/bin/src/main.rs

```diff
@@ -0,0 +1,46 @@
+//! Scroll binary
+#![cfg(all(feature = "scroll", not(feature = "optimism")))]
+
+use clap::Parser;
+use reth_node_builder::{engine_tree_config::TreeConfig, EngineNodeLauncher, Node};
+use reth_provider::providers::BlockchainProvider2;
+use reth_scroll_cli::{ScrollChainSpecParser, ScrollRollupArgs};
+use reth_scroll_node::ScrollNode;
+
+#[global_allocator]
+static ALLOC: reth_cli_util::allocator::Allocator = reth_cli_util::allocator::new_allocator();
+
+fn main() {
+    reth_cli_util::sigsegv_handler::install();
+
+    // Enable backtraces unless a RUST_BACKTRACE value has already been explicitly provided.
+    if std::env::var_os("RUST_BACKTRACE").is_none() {
+        std::env::set_var("RUST_BACKTRACE", "1");
+    }
+
+    if let Err(err) = Cli::<ScrollChainSpecParser, ScrollRollupArgs>::parse().run(
+        |builder, rollup_args| async move {
+            let engine_tree_config = TreeConfig::default()
+                .with_persistence_threshold(rollup_args.persistence_threshold)
+                .with_memory_block_buffer_target(rollup_args.memory_block_buffer_target);
+            let node = ScrollNode;
+            let handle = builder
+                .with_types_and_provider::<ScrollNode, BlockchainProvider2<_>>()
+                .with_components(node.components_builder())
+                .launch_with_fn(|builder| {
+                    let launcher = EngineNodeLauncher::new(
+                        builder.task_executor().clone(),
+                        builder.config().datadir(),
+                        engine_tree_config,
+                    );
+                    builder.launch_with(launcher)
+                })
+                .await?;
```

Comment:
remark: Maybe we should add a TODO or create an issue to consider adding support for the legacy engine.

Reply:
do we need to support the legacy engine? I explicitly removed it thinking there wasn't a need for it.

Reply:
I think there is a functional difference between using the new engine and the legacy engine. I think (but may be wrong) the functional difference is related to subscriptions to the chain head being delayed in the new engine. Lets ignore the legacy engine for new and create an issue to consider if we need to support it.

Reply:
I see, I have not looked into the legacy engine much, will raise the issue.

Reply:
added https://github.com/scroll-tech/reth/issues/73